### PR TITLE
[Bugfix] On deleting any child widget from the listview widget, removes the widget from the editor and from the currentState

### LINKED ISSUE #3680: On deleting any child widget from the Listview widget, removes the widget from the editor but Left Sidebar Inspector displays the widget and its data.
### What is the expected behaviour?
Deleting child widgets from the **Listview widget** should remove the widget from the editor and the changes(removable of the widget) should be available through the editor's state

### What is the current behaviour?
On deleting any child widget from the listview widget, removes the widget from the editor but Left Sidebar Inspector displays the widget and its data.

### How to reproduce the issue?

1. Add a Listview widget from the widget manager
2. Drop a few widgets inside the listview
3. Pin the Left sidebar Inspector and start removing the widgets inside the Listview (children)

### Screenshots or Screencast

https://user-images.githubusercontent.com/67645175/181208393-e94b0df3-719c-4317-ab46-a65e9cd1909c.mov

### PR DESCRIPTION
**Pull request by arpitnath**
Resolves #3680

### CODE REVIEW

**Comment by shubh22** [comment]
Task linked: [CU-2me44w5 Listview: deleting any child widget from the listview widget, deletes the widget from the editor but not from the currrentState until we reload the page](https://app.clickup.com/t/2me44w5)

**Review by sherfin94 — CHANGES_REQUESTED** [review]
Putting ListView inside ListView crashes.

File: frontend/src/Editor/Components/Listview.jsx [hunk 1]
```diff
@@ -54,6 +56,39 @@ export const Listview = function Listview({
...
 
+  const renderLimit = useRenderLimit(1, exposedVariables);
+
+  useEffect(() => {
+    if (renderLimit < 1 && !_.isEqual(exposedVariables.data, childrenData)) {
+      setExposedVariable('data', childrenData);
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [exposedVariables]);
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Components/Listview.jsx:67]:
```suggestion
  useEffect(() => {
    if (!_.isEqual(exposedVariables.data, childrenData)) {
      setExposedVariable('data', childrenData);
    }

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [JSON.stringify(exposedVariables)]);
```

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@arpitnath The `exposedVariable` is an object and the reference keeps on changing every time and causing multiple re-renders. Instead of adding a custom hook and counting the number of renders, stringify the `exposedVariable` in the `useEffect` dependency array.

**Comment by kavinvenkatachalam** [comment]
@arpitnath I tried the same as you described in the issue. Seems like `LeftSideBarInspector` is not getting updated after deleting the listItem. Can you check it out?

**Comment by kavinvenkatachalam** [comment]
> Putting ListView inside ListView crashes.

@arpitnath This issue still exists. Can you please take a look?

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
Listview crashes when the listview is dropped into it

**Comment by arpitnath** [comment]
@kavinvenkatachalam Putting this PR on hold for now.
cc: @sherfin94 @shubh22

**Pull request closed without merging**